### PR TITLE
Change fee schedule

### PR DESCRIPTION
**Pull request by mschneider**
* remove SRM/MSRM fee discount
* fee tiers purely based on market ids
* 3 tiers: Base, Flagship, Stable
* 99.9% of fees paid out to maker & gui, only dust reamins

### CODE REVIEW

**Comment by krakong** [comment]
so base taker fee goes from 4bps to 20bps and maker fee from 0bps to 10bps, correct?
what is the reasoning for increasing fees? incentivize better uis?

**Comment by DonDuala** [comment]
Is there any suggested process to categorize future markets? Where is the line drawn from Base to Flagship specifically?

**Comment by mzs-dev** [comment]
> so base taker fee goes from 4bps to 20bps and maker fee from 0bps to 10bps, correct? what is the reasoning for increasing fees? incentivize better uis?

The fee on program is in tenthOfBPS so the fee is now 
0.005% for Stable Pair Markets
0.02% for Flagship Markets,
0.1% for Base Markets (Default)

**Comment by mschneider** [comment]
> Is there any suggested process to categorize future markets? Where is the line drawn from Base to Flagship specifically?

i guess the process is send a PR.

i would prefer, if serum could adopt the uniswap model where you choose when you create the market. it's just not well supported by this version of serum.

**Comment by mschneider** [comment]
This PR proposes a taker fee / maker rebate model, 50% rebate to maker, 50% to gui:

USDT/USDC 1bps 
SOL/USDC 4bps
all other markets 20bps

**Comment by flahpah** [comment]
0.2% fees are egregious. most of the onchain swap protocols are 0.03% or less. worst binance/bybit spot fee rate tiers are 0.1% - which subsidizes their entire operating costs.  

Should align them to be at least competitive with the rest of solana defi.

**Comment by jkbpvsc** [comment]
What if the gui chooses to forgo fees?
Or the trader isn't using a gui to trade?

**Comment by DonDuala** [comment]
> 

The attempt here is to make fees higher on less liquid tokens similar to uniswap's model:
image

Flagship fees are 4bps and would be competitive with most cex's on the majors. You want to over-incentivize liquidity on the less liquid pairs by offering more fees for makers & referrers

**Comment by mschneider** [comment]
> What if the gui chooses to forgo fees? Or the trader isn't using a gui to trade?
> 
> I agree 20bps is a steep fee, would like to see the reasoning here.

uis should be able to forgo the fees, it's their choice.
traders who don't use a gui just self-refer and save 50%.
0.2% is cheaper than uniswap or orca.

File: dex/src/fees.rs [hunk 1]
```diff
@@ -83,61 +75,44 @@ const fn fee_tenth_of_bps(tenth_of_bps: u64) -> U64F64 {
...
-            () if srm_held >= one_srm * 1_000_000 => FeeTier::SRM6,
-            () if srm_held >= one_srm * 100_000 => FeeTier::SRM5,
-            () if srm_held >= one_srm * 10_000 => FeeTier::SRM4,
-            () if srm_held >= one_srm * 1_000 => FeeTier::SRM3,
-            () if srm_held >= one_srm * 100 => FeeTier::SRM2,
-            () => FeeTier::Base,
+        FeeTier::Base
+    }
+
+    fn maker_rate(self) -> U64F64 {
```
Comment (ckamm) [inline dex/src/fees.rs:92]: I've seen this misinterpreted as a maker fee. Rename to `maker_rebate_bps()` and the other to `maker_rebate_qty()`?

**Comment by ckamm** [comment]
With this proposal the trade volume could be arbitrarily inflated at nearly no cost, since wash trading incurs no fees when self-referring. It would be nice to have sensible stats. Maybe the referrer rebate could be one less bps, to at least have a small guaranteed cost? The program could burn that part.

**Comment by jkbpvsc** [comment]
Should we look into subsidizing cranking cost from fees at this point?

**Comment by solserer-labs** [comment]
just let the creator of the market decide the fees? makers can then just choose which one suits them best..

File: dex/src/fees.rs (hunk 1 above)
Comment (mschneider) [reply dex/src/fees.rs:92]: I wanted to keep it in sync with the taker functions. Will change both in that case

**Review by kizmt — APPROVED** [review]

**Comment by crispheaney** [comment]
.2% taker fee for markets like btc/eth seems too aggressive imo. AMMs are forced to have higher fee because they don't have spread to dynamically add a fee. with a smaller taker fee, more of the effective fee can be priced dynamically.

**Comment by skynetcap** [comment]
I agree, 0.2% taker fee for "other" markets is too high and should be massively reduced.

**Review by robre — APPROVED** [review]
looks fine

**Comment by SpaceMonkeyForever** [comment]
Agree with ckamm that if (rebate + self-refer) means full rebate to maker (which as far as I can see is the case) can lead to wash trading.

I also wanted to say that the current fee model (0 bps maker, 4 bps taker) works fine for me (I'm marking making). Not to say that the maker rebate won't be better for me, but I don't need that to continue quoting.

**Comment by davidkzeng** [comment]
(comment from former Mango/Serum MM). Agree with SpaceMonkey's comment.

A -10bps maker rebate for non flagship markets forces MM's to quote a spread of at least 20 bps. IIRC, spreads lower than 20bps on Serum were a reasonably common occurrence on BTC-MSOL-SRM/USDC markets, especially when volatility was low. From a MM incentive perspective, most MM's would rather have the option to quote tighter and take more order flow that would have gone to AMMs.

AMM's need larger fees on more exotic markets due to a higher toxic to non-toxic flow ratio, thus needing to collect more fees from the non-toxic flow. This effect is much less pronounced for the CLOB model, where an MM could just incorporate external prices/volatility into their price and spread to greatly reduce the amount of toxic flow they're exposed to.

**Comment by mschneider** [comment]
we can easily change the fees in a follow-up pr, i think it'd be best to discuss this as a separate issue, so the conversation isn't hidden in this closed PR. also come to the community meetup on discord 12/1 10pm UTC

**Comment by infraraydium** [comment]
I've been thinking a lot about fees over and how rebates would help Raydium spreads. I don't have a "Twitter ready" solution yet, but I generally agree with davidkzeng's second comment. Regardless of the pool, Raydium's AMM places its **first** bid/ask where max_bid = pool_mid_price*(1-(trade_fee + spread)) and min_ask = pool_mid_price*(1 + (trade_fee + spread)).

Where trade_fee = 25bps for swaps and the spread can vary, usually accounting for slippage (5bps), with the end result of the first bid/ask 30bps away from the mid-price of the pool. The spread was increased to 25 when consumption issues on Serum's side came up a few months ago. 

Raydium's AMM is unlikely to be a competitive maker in high-volume, tightly-quoted markets but it can be effective filling books as a lone market maker in low-liquidity, lower-volume markets for longer-tail assets. A 10 basis point maker rebate would help cut the first bid/ask on any market to 20bps away from the pool mid-price. Though, I'm not sure it matters in the end? 

I believe a GUI rebate is important to help hosters offset RPC costs.

**Comment by 0xSoju** [comment]
I echo @mschneider's sentiments that the we don't have to worry about the various fee tiers for now. Our priority should be getting this out to drive MM and UI uptake.

Let's focus on shipping the new fees (and the fee tier) but also setting **more** markets to Flagship. My suggestion would be to apply Flagship fees to SOL-USDC, mSOL-USDC, stSOL-USDC and RAY-USDC. This way, we can ensure that MMs like Spacemonkey will face no issues quoting on his markets while also testing out/observing how the Base fee tier works for low-liquidity markets. These markets are vital to Openbook and will drive most of the volume, and we can afford to explore with the others. 

My biggest concern is that if mSOL and stSOL is under Base with 20bps, it will lose Jup volume to Lifinity or Orca. 

Alternatively, ship everything as Flagship (other than USDC/USDT)

**Comment by SpaceMonkeyForever** [comment]
@0xSoju Btw some of the tokens that you didn't include may have more volume once I start quoting there e.g. ETH.

Sorry to keep repeating this, but I really think the idea of having different tiers *per market* at all is a very bad idea and doesn't make any sense to apply to an exchange. It makes sense for AMMs because they are market makers, but exchanges are just a platform for market makers. Put another way, if an AMM would start quoting on an exchange (like Raydium I think), they would choose different fee (spread) per market, but not the exchange itself.

If we want different tiers, I think volume-based (basically cheaper for MMs) or holdings-based (SRM, MNGO, FTT, etc) is the right, and typical, way.

**Comment by SpaceMonkeyForever** [comment]
Just to be clear, if we want to increase volume for illiquid markets, we could *decrease* taker fee (rather than increase it) or to be more precise decrease (taker + maker) fee, so e.g. decrease taker or increase maker rebate (negative fee). So, it could make sense to have different fees per market but as long as we keep the sum of (taker+maker) low to not drive away the volume, rather than increase the sum.

**Comment by 0xSoju** [comment]
As per our Community Townhall on Monday 4pm UTC, a slight conclusion was reached. 

We have two options ahead of us right now:
1. Introduce a flat 4bps fee for all markets (Except USDC/USDT), and provide 50% rebates to Makers, and 50% rebates to UIs.
2. Introduce a flat 2bps fee for all markets (except USDC/USDT), and provide 100% of it to the UI. MMs will quote at their preference to capture the bips they prefer via spreads. 

Details can be read more here: https://docs.google.com/document/d/1mLLoLBSCTfuZj2XpU5M632jLYTv9o7G2kQ-zpE5eO_s/edit

Going forward, more research and clarity is needed on what drives MMs to use Openbook. I am leading an effort to discuss with market makers to achieve a good model, and welcome discussions in Discord in a contributors-only chat. If you are interested in joining this effort, please ping me on Discord. 

Max aims to introduce this new schedule by EOW.

**Comment by SpaceMonkeyForever** [comment]
@0xSoju I vote for option 1, keep the 4 bps taker fee and split it between UI and maker.

Even though I may have sounded like I'm against maker rebates on the call, it's only because you wanted to increase taker fee too much.

Option 1 is good because:
- Taker fee stays low, we need to be competitive on jup.ag to get retail flow. Retail doesn't care about fees but jup.ag does and most people (like me) use it.
- 2 bps maker rebate helps against toxic taker flow
- Keeps the "effective spread" (final bps paid by taker = fee+spread) flexible because it's possible to quote 2 bps tighter to achieve the same effective spread as Option 2. That's not the case if e.g. taker/maker >= +/- 20 bps, can't quote 20 bps tighter when I'm currently quoting 10 bps per side, it would cross the spread.


I really hope that the 2 bps UI rebate (which is about 200$/day for 1 million volume) helps bring a decent UI to openbook because that's a major problem right now. Solape is very buggy unfortunately and would scare retail away (funds not settled, e.g.), openserum is only for viewing and not trading, dex.marginfi has some issues too e.g. no balance is not visible.

**Comment by siong1987** [comment]
@SpaceMonkeyForever added a PR for option 1 here #19.

**Pull request merged**